agent/util-scripts/pbench-tool-meister-stop

```diff
@@ -38,22 +37,33 @@ def is_running(pid):
     return True
 
 
-def main(argv):
-    """Main program for the tool meister stop.
+def main(_prog, parsed):
+    """Main program for the tool meister stop CLI interface.
+
+    Stopping the Tool Meisters involves four steps:
+
+    1. End ("end") the run of the persistent tools
 
-    This simply sends the "terminate" message to the redis server so that all
-    connected services, tool-meisters, tool-data-sink, etc. will shutdown.
-    Once all services acknowledge the receipt of the "terminate" message, we
-    declare victory.
+    2. Collect any requested system configuration information ("sysinfo")
 
-    TBD:
+    3. Record in the local metadata.log file all relevant metadata from the
+    stop operation, including the timing to end the persistent tools, the
+    timing to collect the system information, and any other data regarding the
+    outstanding state of the Tool Meisters.
 
-    We currently have a mode to "double-check" that the services all stopped
-    by ssh'ing into all the local/remote hosts and inspecting the system for
-    any lingering processes.
+    4. Sends the "terminate" message to the redis server so that all connected
+    services, tool-meisters, tool-data-sink, etc. will shutdown.  Once all
+    services acknowledge the receipt of the "terminate" message, we look to
+    ensure that the local Tool Data Sink and optional Tool Meister instance
+    have shutdown by verifying their recorded pids are no longer seen on the
+    local host.
+
+    FIXME: previously we had a mode to "double-check" that the services all
+    stopped by ssh'ing into all the local/remote hosts and inspecting the
+    system for any lingering processes.  Should we offer a mode like that?
```

Comment:
From the CLI interface point of view one can argue that it is, because the CLI still uses an SSH to the remote hosts to orchestrate the initial creation of the remote Tool Meisters.  This is why this comment is here, because we removed that code and now don't have any analogous method to SSH to remote hosts and ensure they Tool Meisters are stopped.

It is a FIXME because we are still make sure this this stop mechanism works, and once we have enough confidence we can probably just drop it.

Reply:
Filed issue #2120 to track this `FIXME`.